### PR TITLE
Feat: add Data source for self-hosted-agent helm values

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

The first part of #450.
In the PR I implemented in the API call and added unit tests + minor refactoring.
A follow-up PR will implement the data source itself.

### CODE REVIEW

File: client/agent.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package client
+
+type Agent struct {
+	AgentKey string `json:"agentKey"`
+}
+
+func (client *ApiClient) Agents() ([]Agent, error) {
```
Comment (TomerHeber) [inline client/agent.go:7]: `Agents()` is not a new code. Moved it to here from agent_project_assignment.

File: client/agent.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	var result []Agent
+	err = client.http.Get("/agents", map[string]string{"organizationId": organizationId}, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (client *ApiClient) AgentValues(id string) (string, error) {
```
Comment (TomerHeber) [inline client/agent.go:22]: This is new.

File: client/agent_project_assignment.go [hunk 3]
```diff
@@ -9,10 +9,6 @@ type ProjectsAgentsAssignments struct {
 	ProjectsAgents map[string]interface{} `json:"ProjectsAgents"`
 }
 
-type Agent struct {
```
Comment (TomerHeber) [inline client/agent_project_assignment.go:12]: moved to above file.

File: client/agent_project_assignment.go [hunk 4]
```diff
@@ -41,18 +37,3 @@ func (client *ApiClient) ProjectsAgentsAssignments() (*ProjectsAgentsAssignments
 
 	return &result, nil
 }
-
-func (client *ApiClient) Agents() ([]Agent, error) {
```
Comment (TomerHeber) [inline client/agent_project_assignment.go:45]: moved to above file.

File: client/agent_project_assignment_test.go [hunk 5]
```diff
@@ -136,67 +136,4 @@ var _ = Describe("Agent Project Assignment", func() {
 			})
 		})
 	})
-
-	Describe("Agents", func() {
```
Comment (TomerHeber) [inline client/agent_project_assignment_test.go:140]: test moved to agent_test

File: client/agent_test.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+				Expect(err).To(Equal(errorMock))
+			})
+
+			It("Should not return results", func() {
+				Expect(actualResult).To(BeNil())
+			})
+		})
+	})
+
+	Describe("AgentValues", func() {
```
Comment (TomerHeber) [inline client/agent_test.go:78]: These tests are new.

File: client/agent_test.go [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+					Get("/agents/"+agentId+"/values", nil, gomock.Any()).
+					Return(errorMock)
+
+				_, err = apiClient.AgentValues(agentId)
+			})
+
+			It("Should fail if API call fails", func() {
+				Expect(err).To(Equal(errorMock))
+			})
+		})
```
Comment (Wassap124) [inline client/agent_test.go:122]: maybe also validate that empty string is returned?

Comment (TomerHeber) [reply client/agent_test.go:124]: @Wassap124 - e8a9727b865e1bcc2a4dba02527c53ec8e1f7e5f

**Review by yaronya — APPROVED** [review]

**Pull request merged**